cmd/root.go

```diff
@@ -50,30 +52,94 @@ func Execute() {
 
 func init() {
 	rootCmd.Flags().StringP(countryCode, "c", "", "Country code in ISO 3 character format")
+	viper.BindPFlag(countryCode, rootCmd.Flags().Lookup(countryCode))
 }
 
 func getGridIntensityForCountry(countryCode string) error {
-	if countryCode == "" {
-		// No country code provided so try to detect it from the user's locale.
-		tag, err := locale.Detect()
+	result, err := ember.GetGridIntensityForCountry(countryCode)
+	if err != nil {
+		return err
+	}
+
+	bytes, err := json.MarshalIndent(result, "", "\t")
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(string(bytes))
+	return nil
+}
+
+// getCountryCode prompts the user to enter a country code. We try to detect
+// a country code from the user's locale but the user can enter another value.
+func getCountryCode() (string, error) {
+	tag, err := locale.Detect()
+	if err != nil {
+		return "", err
+	}
+
+	region, _ := tag.Region()
+	country := region.ISO3()
+
+	fmt.Printf("Provider %s needs an ISO country code as a region parameter.\n", ember.Provider)
+	if country != "" {
+		fmt.Printf("%s detected from your locale.\n", country)
+	}
+
+	var reader = bufio.NewReader(os.Stdin)
+	country, err = reader.ReadString('\n')
+	if err != nil {
+		return "", err
+	}
+
+	return strings.TrimSpace(country), nil
+}
+
+func runWithError() error {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return nil
+	}
+
+	configFile := filepath.Join(homeDir, configDir, configFileName)
+	viper.SetConfigFile(configFile)
+
+	err = viper.ReadInConfig()
+	if _, ok := err.(*fs.PathError); ok {
+		// Create config dir if it doesn't exist.
+		err = os.Mkdir(filepath.Join(homeDir, configDir), os.ModePerm)
 		if err != nil {
 			return err
 		}
 
-		region, _ := tag.Region()
-		countryCode = region.ISO3()
+		// Create config file if it doesn't exist.
+		_, err = os.Create(configFile)
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
 	}
 
-	result, err := ember.GetGridIntensityForCountry(countryCode)
+	country := viper.GetString(countryCode)
+	if country == "" {
+		country, err = getCountryCode()
+		if err != nil {
+			return err
+		}
+
+		viper.Set(countryCode, country)
+	}
+
+	err = getGridIntensityForCountry(country)
```

Comment:
Do you think it's worth making explicit that the country code is being used to make the request/lookup against our set of country / carbon intensity codes?

I know the returned JSON blob mentions that the country and country code, but for the first time, when I looked over it, I felt like having a short one line outlining what was about to happen before showing the returned value would make it easier to understand what was happening. 

This probably isn't needed on subsequent uses, but I think it provides a more gentle way into this domain.

Reply:
Yes, I agree I think we should explain more what is going on.

I've been thinking we could add an `--output` flag that defaults to text but also supports JSON.
The text output could walk the user through choosing a provider and region. As well as linking to our developer site for more docs. WDYT?

Reply:
yes. would you mind creating an issue for this? I think this would be a good one to include Fershad for this, so he can create some tutorials and add some reference to it in the developers.tgwf  site

Reply:
@mrchrisadams I created #32 for this. I agree it would be good to work with Fershad on this and to have some tutorials.